### PR TITLE
Lord-Giganticus's Takochu fixes

### LINKED ISSUE #13: mFilesystem is not closed when closing the RARC explorer
**Describe the bug**
mFilesystem is not closed when closing the RARC explorer form.

**Game**
Both

**To Reproduce**
Open the RARC explorer, open a RARC, close the RARC explorer, open the BCSV editor, open the same RARC.

**Expected behaviour**
self explanatory but anyway, mFilesystem needs to be closed when closing the rarc explorer form 

**Desktop (please complete the following information):**
 - OS: Windows 10

### PR DESCRIPTION
**Pull request by Lord-Giganticus**
# Changes made to Takochu:
- Made the references to GL_EditorFramework, and SuperBMDLib actually decent. ~~I had to add Spotlight cuz GL uses it for the Fasterflect package.~~
- Fixed #13 
- Added use of [WindowsAPICodePack-Core](https://www.nuget.org/packages/WindowsAPICodePack-Core/), [WindowsAPICodePack-Shell](https://www.nuget.org/packages/WindowsAPICodePack-Shell/), and [WindowsAPICodePack-ShellExtensions](https://www.nuget.org/packages/WindowsAPICodePack-ShellExtensions/) to make the FolderBrowserDialog not hurt everyone's eyes with non-vista dialogs.
- ~~Added a icon (I got jebated ngl)~~
- ~~Changed the .gitignore to [Github's Visual Studio .gitignore](https://github.com/github/gitignore/blob/master/VisualStudio.gitignore) to avoid oddities~~
- Updated the README to have basic instructions on how to build it.

### CODE REVIEW

File: Takochu/ui/RARCExplorer.cs [hunk 1]
```diff
@@ -67,12 +67,11 @@ private void ExportFilesAndDirs(TreeNode node, string parent, string folderPath)
             }
         }
 
-        private RARCFilesystem mFilesystem;
+        public RARCFilesystem mFilesystem;
```
Comment (Lord-Giganticus) [inline Takochu/ui/RARCExplorer.cs]: Gonna change this back to private, I made it public before realizing how to properly fix #13

Comment (Lord-Giganticus) [reply Takochu/ui/RARCExplorer.cs]: Changed back.

**Comment by shibbo** [comment]
So a few things to do before I merge:
- Remove the icon as I already have one
- No reason for the gitignore change, as 99% of it will never apply to this project
- Remove spotlight dependency, as I feel this is a waste of space for a simple reference

File: README.md [hunk 2]
```diff
@@ -1,2 +1,13 @@
 # Takochu
 An editor for Super Mario Galaxy 1 and 2.
+# Building from Source
+Requirements:
+- [Microsoft Visual Studio 2019](https://visualstudio.microsoft.com/) for building.
+- The .NET Framework 4.7.2 Targeting Pack (Can be found in Visual Studio by clicking Tools, Get Tools and Features, Individual Components, .NET Framework 4.7.2 Targeting Pack)
+- [gitforwindows](https://gitforwindows.org/) (It makes life nicer)
+
+Cloning:
+To clone this repo either use Visual Studio's git repo extension to clone the repo properly (Will include the submodules.) or use `git clone https://github.com/Lord-Giganticus/Takochu.git --recursive` (THE URL IS DIFFERENT ON OTHER FORKS/THE MASTER REPO)
```
Comment (shibbo) [inline README.md]: by default the URL should really point to here

File: Takochu.sln [hunk 3]
```diff
@@ -5,16 +5,182 @@ VisualStudioVersion = 16.0.30114.105
...
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Takochu", "Takochu\Takochu.csproj", "{2B69F139-E33C-4D77-AC4D-2C73923902BC}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Gl_EditorFramework", "References\GL_EditorFramework\Gl_EditorFramework\Gl_EditorFramework.csproj", "{D593A72D-554D-40FB-9967-503CB15A744F}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "SuperBMDLib", "References\SuperBMD\SuperBMDLib\SuperBMDLib.csproj", "{828BE5E5-9E98-46C9-B63E-D2D03322A825}"
+EndProject
 Global
 	GlobalSection(SolutionConfigurationPlatforms) = preSolution
+		Debug Moonlight|Any CPU = Debug Moonlight|Any CPU
+		Debug Moonlight|x64 = Debug Moonlight|x64
```
Comment (shibbo) [inline Takochu.sln:15]: any way to remove these?

**Pull request merged**